modules/book/book.module

```diff
@@ -545,11 +545,13 @@ function book_prev($book_link) {
     return NULL;
   }
   $flat = book_get_flat_menu($book_link);
-  // Assigning the array to $flat resets the array pointer for use with each().
+  reset($flat);
```

Comment:
This is interesting. I don't think this is necessary, since the comment which you're removing explicitly states that it isn't. (And the example in https://www.php.net/manual/en/function.current.php) also implies that it isn't.) The only thing that would need to be changed is strip the " for use with each()" off the end of the comment.

I can totally understand writing explicit code like reset() calls, so we're not dependent on half-documented internal PHP features. I do it myself too.
...But... I'm still commenting because I notice the PR is inconsistent :) No reset() was added in the above menu.inc code either. That may be a reason to not do it here either.

Reply:
I read the D7 patch again, and
* andypost explicitly argues for including a reset(), in https://www.drupal.org/project/drupal/issues/2925449#comment-12380008. \*shrugs\* 
* then says "other places are still missing it too" in # 21, which leads to a boatload of confusion and wrong patches in the rest of the issue because people are reset()s in the wrong places.

Sigh. Never mind, let's not be consistent, let's just commit the D7 patch version.